subprojects/core/src/main/java/org/gradle/execution/plan/DefaultNodeValidator.java

```diff
@@ -51,6 +71,18 @@ public boolean hasValidationProblems(LocalTaskNode node) {
                     .withUserManual(userManualReference.getId(), userManualReference.getSection())
                     .nagUser();
             });
-        return !problems.isEmpty();
+    }
+
+    private void reportErrors(List<TypeValidationProblem> problems, TaskInternal task, WorkValidationContext validationContext) {
+        ImmutableSet<String> uniqueErrors = problems.stream()
+                .filter(problem -> !problem.getSeverity().isWarning())
+                .map(TypeValidationProblemRenderer::renderMinimalInformationAbout)
+                .collect(ImmutableSet.toImmutableSet());
+        if (!uniqueErrors.isEmpty()) {
+            throw WorkValidationException.forProblems(uniqueErrors)
+                    .limitTo(WorkValidationException.MAX_ERR_COUNT)
```

Comment:
Would it make sense to have a `limitToDefault()` method on the builder, so you wouldn't need to expose the constant from `WorkValidationException`?

Reply:
Was able to make the max errors constant `private` by making the summarizer a static inner class of the exception and using the limit as a default when constructing a new builder - this allowed completely removing the need to call `limitTo()`.